tests/unit/package_managers/yarn/test_main.py

```diff
@@ -115,3 +120,43 @@ def test_fetch_dependencies(mock_yarn_cmd: mock.Mock, rooted_tmp_path: RootedPat
     )
 
     assert str(exc_info.value) == "berryscary"
+
+
+@pytest.mark.parametrize(
+    "is_zero_installs",
+    (
+        pytest.param(True, id="zero-installs-project"),
+        pytest.param(False, id="regular-workflow-project"),
+    ),
+)
+@mock.patch("cachi2.core.package_managers.yarn.project.YarnRc.to_file")
+def test_set_yarnrc_configuration(mock_to_file: mock.Mock, is_zero_installs: bool) -> None:
+    yarn_rc = YarnRc(RootedPath("/tmp/.yarnrc.yml"), {})
+
+    project = mock.Mock()
+    project.is_zero_installs = is_zero_installs
+    project.yarn_rc = yarn_rc
+
+    output_dir = RootedPath("/tmp/output")
+
+    _set_yarnrc_configuration(project, output_dir)
+
+    expected_data = {
+        "checksumBehavior": "throw",
+        "enableImmutableInstalls": True,
+        "enableScripts": False,
+        "enableStrictSsl": True,
+        "enableTelemetry": False,
+        "ignorePath": True,
+        "unsafeHttpWhitelist": [],
+        "plugins": [],
```

Comment:
Humm, good catch. After quickly reading the docs, I think it might be worth setting it too. 

The reason we're setting all props even if they have a default is to make sure we override any potential user config.

Reply:
@brunoapimentel  Soo, in context of the code review, do we have an agreement that we want the pnpMode option, or do we need more discussion about that? I'm only asking whether I need to wait for it or hit approve on this.

Reply:
Yeah, I believe it's worth setting `pnpMode`. I'll update this PR.

Reply:
Done.